class.jetpack-signature.php

```diff
@@ -230,12 +241,27 @@ function normalized_query_parameters( $query_string ) {
 		return $pairs;
 	}
 
-	function encode_3986( $string ) {
-		$string = rawurlencode( $string );
-		return str_replace( '%7E', '~', $string ); // prior to PHP 5.3, rawurlencode was RFC 1738
+	function encode_3986( $string_or_array ) {
+		if ( is_array( $string_or_array ) ) {
+			$encoded = array();
+			foreach ( $string_or_array as $key => $value) {
+				$encoded[ $key ] = $this->encode_3986( $value );
+			}
+			return $encoded;
```

Comment:
You could `return array_map( array( $this, 'encode_3986' ), $string_or_array );` for more FP/Calypso style instead of the temp var and `foreach` (L246-L250), if I'm not mistaken. (You could also make `encoude_3986` static.)

Reply:
Nice, thanks! Simplified in https://github.com/Automattic/jetpack/pull/8461/commits/e7d329f7a2f777b27030590e787300a7943a415c.